lib/srv6.c

```diff
@@ -114,3 +125,53 @@ const char *seg6local_context2str(char *str, size_t size,
 		return str;
 	}
 }
+
+struct srv6_locator *srv6_locator_alloc(const char *name)
+{
+	int namelen = 0;
+	struct srv6_locator *locator = NULL;
+
+	locator = XCALLOC(MTYPE_SRV6_LOCATOR, sizeof(struct srv6_locator));
+	if (locator) {
+		namelen = sizeof(locator->name);
+		if (namelen > SRV6_LOCNAME_SIZE - 1)
+			namelen = SRV6_LOCNAME_SIZE - 1;
+		strlcpy(locator->name, name, namelen);
+		locator->current = 0;
+		locator->functions = list_new();
+		QOBJ_REG(locator, srv6_locator);
+	}
+	return locator;
+}
+
+void srv6_locator_free(struct srv6_locator *locator)
+{
+	XFREE(MTYPE_SRV6_LOCATOR, locator);
```

Comment:
This leaks the list allocated on line 141

Reply:
I fixed it